synapse/handlers/auth.py

```diff
@@ -205,15 +205,23 @@ def __init__(self, hs: "HomeServer"):
         # type in the list. (NB that the spec doesn't require us to do so and
         # clients which favour types that they don't understand over those that
         # they do are technically broken)
+
+        # start out by assuming PASSWORD is enabled; we will remove it later if not.
         login_types = []
```

Comment:
Looks like `login_types` really wants to be a `set` so we can just add things to it arbitrarily without having to check first.

Reply:
yes, but see the comment just above.

Reply:
Right, I suppose there's other ways to solve that, but no need to restructure the whole thing.